### ISSUE TITLE
Windows is missing

### ISSUE THREAD

**Issue by andrejohansson** [saltstack-formulas/salt-formula#177]
How about salting/upgrading salt minions on windows in a stable way? I cant see support for that in this formula.

I have some work implemented for this but its very "hacky" due to that nssm.exe often hangs and locks files after an uninstall. To come around this I'm rebooting the server and continue the installation afterwards but this is far from optimal. 

I can however, share the work if anyone is interested.

**Comment by gravyboat** [comment]
Let's ask @UtahDave. Dave what's your opinion on Windows stuff in this formula (especially if it is particularly hacky)? I'm thinking it would be okay but should be in a separate state from the standard stuff just so we don't have tons of if statements and such.

**Comment by TheBigBear** [comment]
I think as of 2015.8.3 release and the way it now cleanly adds a separate local job to actually schedule and run the minion updates, this should now be fairly straight-forward. ( but to keep it really 'simple' would probably need conditionals to pin it to Beryllium >= 2015.8.3 )

**Comment by andrejohansson** [comment]
@TheBigBear, I know that in later versions, when you run for example "service.restart salt-minion" this creates a scheduled task that restarts the minion itself. Is this what you mean in the upgrade process? Because I have found that there is still some oddities with nssm.exe getting stuck and the installer still says that it removed everything.

**Comment by UtahDave** [comment]
I'm OK with having Windows stuff here. It's mostly a matter of trying to do it cleanly. If it gets too messy it might be better to have a similar but separate formula for just Windows.

@andrejohansson do you have a PR we could look at?

**Comment by iamseth** [comment]
I added pull #210 

It's very minimal but it's been working for the past month for us.

I use the windows software repo to handle upgrading minions and the changes to map.jinja allow me to configure the minions identically for our Linux and Windows hosts. Here's what our basic pillar looks like for minions which works on both types of minions today.

```yaml
salt:
  minion:
    master:
      - ueinf-dvslt0001.sanitized.example.com
      - ncinf-dvslt0001.sanitized.example.com
```

**Comment by iamseth** [comment]
I should mention that I've only tested this with Windows 2012 R2 minions running in Azure.

**Comment by gravyboat** [comment]
I merged in #210, thanks @iamseth
